lampod/src/ln/inventory_manager.rs

```diff
@@ -39,6 +40,20 @@ impl InventoryHandler for LampoInventoryManager {
                 let (_, height) = self.channel_manager.onchain.backend.get_best_block()?;
                 let blockheight = height.unwrap_or_default();
                 let lampo_dir = self.channel_manager.conf.root_path.to_string();
+                // We provide a vector here as there may be other types of address in future like tor and ipv6.
+                let mut address_vec = Vec::new();
+                let address = self.channel_manager.conf.announce_addr.clone();
+                if address != None {
+                    // Safe to unwrap
+                    let addr = address.unwrap();
```

Comment:
```suggestion
                if if let Some(addr) = address {
```

Please use the good feature of rust

Reply:
Sorry for that